File: Space Invaders C#/Assets/Scripts/SpaceInvaders/EnemyColliders.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class EnemyColliders : MonoBehaviour
{
    //public EnemyMovement enemyMovement;
    void Start()
    {

    }
    void OnTriggerEnter(Collider col)
    {

        if (gameObject.tag != "Boss")
        {
            if (col.gameObject.name == "Left_Barrier")
            {
                EnemyMovement.goLeft = false;
                EnemyMovement.goDown = true;
            }
            else if (col.gameObject.name == "Right_Barrier")
            {
                EnemyMovement.goLeft = true;
                EnemyMovement.goDown = true;
            }
            else if (col.gameObject.name == "Player")
            {
                Database.hitPlayer();
            }
            else if (col.gameObject.tag == "Barrier")
            {
                Destroy(col.gameObject);
            }
        }
    }
}

File: Space Invaders/Assets/Scripts/SpaceInvaders/EnemyColliders.js
#pragma strict

var enemyMovement: EnemyMovement;

function OnTriggerEnter(col: Collider) {

    if (gameObject.tag != "Boss") {
        if (col.gameObject.name == "Left_Barrier") {
            enemyMovement.goLeft = false;
            enemyMovement.goDown = true;
        }
        else if (col.gameObject.name == "Right_Barrier") {
            enemyMovement.goLeft = true;
            enemyMovement.goDown = true;
        }
        else if (col.gameObject.name == "Player") {
            Database.hitPlayer();
        }
        else if (col.gameObject.tag == "Barrier") {
            Destroy(col.gameObject);
        }
    }
}
